### ISSUE TITLE
Release 1.0.0

### ISSUE THREAD

**Issue by ericsink** [ericsink/SQLitePCL.raw#94]
This is a tracking issue for the remaining things I need to do for the 1.0.0 release.

## Remaining issues

- New docs
- UWP test apps have a problem with clashing ids for deploy?
- Combo lib package to solve the net45 -> win/mac/linux problem
- Fix push.ps1

## What's new in 1.0.0?

In terms of the code, not much.  But the builds and packaging have changed a lot.  

At a very high level, this is the release that happened because Android N fouled everything up.

At a slightly-less-high level, the fixes and new features are:

- All package names now begin with SQLitePCLRaw.  For example, the main package is now called SQLitePCLRaw.core.
- Support netstandard (#90)
- Switch all the tests to xUnit.net
- Fix problems with UWP apps failing validation (#85)
- Support winsqlite3.dll (the one built-in to Windows 10)
- Fix and refactor things where Android N broke my old assumptions
- Provide an easier way for people to use custom sqlite3 builds (#71)
- Improve my automated tests to test the nuget packaging, not just the code
- Make the iOS sqlite builds compatible with iOS 8 (#82)
- Be compatible with strong naming (#93 and #53)
- sqlite3_threadsafe (#92)
- Update e_sqlite3 (the sqlite builds which I bundle) to 3.14.1

## Packaging changes

First, a review of where we've been.

In 0.8, SQLitePCL.raw was a big nuget package which contained everything.  MSBuild properties were used to select different implementations for cases where somebody wanted to use sqlcipher or whatever.  That was becoming intolerable.

In 0.9.x, the native code was moved out into separate packages.  We had SQLitePCL.plugin.\* packages, which were needed in certain cases.  We had SQLitePCL.native.\*, which were needed in certain cases.  And we had SQLitePCL.bundle_\*, which tried to make things easier again by automatically providing dependencies and initialization as needed.

The key here is that in 0.9.3, the main assembly SQLitePCL.raw contains the implementation for loading the "default sqlite3" library.  And other use cases required another implementation to be injected to replace it.

But in Android N, using the system sqlite3 is not allowed anymore, so the default implementation in the main assembly is always wrong for that case.  And in certain UWP cases, having that implementation present causes problems with validation.  And in the .NET Native case, the extra implementation could cause build failures.  Bottom line:  That built-in default implementation doesn't make sense anymore.

So, in 1.0.0, the main assembly contains no implementation of the ISQLite3Provider interface.  It still contains lots of other stuff, so that it can present a sane and consistent API upward.  But for the purpose of actually talking to a SQLite native library instance somewhere, it requires an implementation of ISQLite3Provider to be injected using raw.SetProvider().

Implementations of those providers are in packages called SQLitePCL.provider.\*.  These are essentially the replacement for SQLitePCL.plugin.\*.  Why the name change?  Because some of the SQLitePCL.plugin.\* packages would have to change meaning.  I figured subtlety would cause more bugs and frustration than a brief time of significant pain.

Note also that provider packages STILL do not contain native sqlite3 bits.  In cases, where those are needed, the packages are named SQLitePCL.lib.\*.

Because of the possibility of breaking changes, the package names have changed for 1.0.  All packages now begin with SQLitePCLRaw.*.  So for example, sqlite-net-pcl should change its dep from SQLitePCL.bundle_green to SQLitePCLRaw.bundle_green, but no other changes should be required in that case.

However, as I understand it, people using 0.9.x will have leftover references to plugin and native packages which are being replaced with provider and lib, respectively.  Those references may need to be manually deleted from their projects.  I may yet be able to find a smoother transition for this issue.

**Comment by ps30** [comment]
Eric - Possible to bring SQLite binaries up to latest 3.14.1 in this release of 0.9.4 (and close issue #86)? I know there are a lot of things going on the list, so if not possible not a problem. We are interested in the "batteries included" windows desktop .Net 4.6.2 build using the SQLite 3.14.x binaries.

**Comment by ericsink** [comment]
Yes, 0.9.4 will have 3.14.1.

E

**Comment by ericsink** [comment]
A pre-release for 1.0 has been pushed to nuget.org.  Note the change in package names.  For example, the package id with 'SQLitePCL.raw' is changing to 'SQLitePCLRaw.core'.

**Comment by fbmiranda7** [comment]
Hi Eric, can we get the issue #85 fixed in pre 1.0 ? the new nuget package id breaks every other dependency, like [sqlite-net-pcl](https://www.nuget.org/packages/sqlite-net-pcl/), which recently was updated to support 0.9.4 changes

**Comment by ericsink** [comment]
> Hi Eric, can we get the issue #85 fixed in pre 1.0 ?

Yes.  This issue is (should be) fixed in the 1.0 prerelease.

> the new nuget package id breaks every other dependency, 

If you mean that every package that depends on SQLitePCL.raw <1.0 will need to be updated to use the new package names, yes, that is true.  I was planning to help/push the other package maintainers to get that done once my 1.0 release is finalized.

Because the 1.0 release contains some minor breaking changes, I wanted to make sure that other package maintainers make the transition explicitly.

> like sqlite-net-pcl, which recently was updated to support 0.9.4 changes

I'm confused by this part, but I think I am just misunderstanding you.  The last update I know about for sqlite-net was in June, and it was to update to 0.9.2, but did not address any of the 0.9.4 stuff, which I no longer call 0.9.4, but now refer to as 1.0.  (It's confusing.)

Bottom line:  Yes, #85 is fixed, and if you want to get that bug fix working with sqlite-net-pcl, then that package needs to be switched over.  I'll talk to @praeclarum next week about getting that done.

**Comment by dhaligas** [comment]
will this work with akavache?

**Comment by ericsink** [comment]
Yes, it will work with Akavache, but last I knew, that project is on 0.8.6, and it will need to be switched over to 1.0, which I will want to help them do, after 1.0 goes final.

I plan to talk with @paulcbetts about getting that done.

**Comment by dhaligas** [comment]
ok great .. we will have to use 1.0.0 for it to work on Android 7

**Comment by bordoley** [comment]
@ericsink: With 1.0.0 will SQLitePCL.raw start honoring semver? Its something to consider when bouncing the major rev of the version number past 0.

**Comment by ericsink** [comment]
Yes, I do plan to follow semver from 1.0 onward.

**Comment by ericsink** [comment]
I just noticed that the following comment had been edited:

> Hi Eric, can we get the issue #85 fixed in some other way (maybe a 0.9.5 release) ? the 1.0.0 new nuget package id breaks every other dependency, like sqlite-net-pcl, which recently was updated to support 0.9.3 changes

Sorry, no.  Fixing this was a major change.  Backporting it to 0.9.x would be unworkable.

I have submitted the pull request to @praelcarum to update sqlite-net-pcl to SQLitePCLRaw 1.0.

**Comment by ericsink** [comment]
Version 1.0 has been released.
